src/robotFactory.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
```

Comment:
I think it would be better to have default coords object, you know how it will look like 😄 
```suggestion
  constructor(name, weight, coords = {z: 0, y:0}, chipVersion) {
```